app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala

```diff
@@ -27,14 +28,34 @@ trait XmlWriter[A] {
 object XmlWriterInstances {
 
   implicit val previousDocumentWriter: XmlWriter[PreviousDocument] = new XmlWriter[PreviousDocument] {
-    override def toXml( value: PreviousDocument ): Option[Elem] = {
+    override def toXml(value: PreviousDocument): Option[Elem] = {
       val categoryCode: Option[Node] = maybeElement("CategoryCode", value.categoryCode)
       val id = maybeElement("ID", value.id)
       val typeCode = maybeElement("TypeCode", value.typeCode)
       val lineNumeric = maybeElement("LineNumeric", value.lineNumeric)
-      val l: List[Node] = List(categoryCode, id, typeCode, lineNumeric).flattenOption
+      val childNodes: List[Node] = List(categoryCode, id, typeCode, lineNumeric).flattenOption
+
+      Option(childNodes).filter(_.nonEmpty).map(nonEmptyChildNodes => <PreviousDocument>{nonEmptyChildNodes}</PreviousDocument>)
+    }
+  }
+
+  implicit val authorisationHolderWriter: XmlWriter[AuthorisationHolder] = new XmlWriter[AuthorisationHolder] {
+    override def toXml(value: AuthorisationHolder): Option[Elem] = {
+      val id = maybeElement("ID", value.identifier)
+      val typeCode = maybeElement("CategoryCode", value.categoryCode)
```

Comment:
Is `CategoryCode` a slightly better name here?

Reply:
fixed now, thanks